Client/Frontend/Widgets/SnackButton.swift

```diff
@@ -0,0 +1,63 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+/**
+ * A specialized version of UIButton for use in SnackBars. These are displayed evenly
+ * spaced in the bottom of the bar. The main convenience of these is that you can pass
+ * in a callback in the constructor (although these also style themselves appropriately).
+ */
+typealias SnackBarCallback = (_ bar: SnackBar) -> Void
+class SnackButton: UIButton {
+    let callback: SnackBarCallback?
+    var bar: SnackBar!
```

Comment:
I don't think it's a good idea to have `bar` be force unwrapped. Either the bar should be part of the initializer (which I'd prefer) or `bar` in `SnackBarCallback` should be optional.